### DISCUSSION TITLE
firefox rejected v0.4.2

### DISCUSSION THREAD

**Discussion by kheina** [kheina-com/Blue-Blocker#294] [Announcements]
Hello,

Your Extension Blue Blocker was manually reviewed by the Mozilla Add-ons team in an assessment performed on our own initiative of content that was submitted to Mozilla Add-ons.

Our review found that your content violates the following Mozilla policy or policies:

- Consent, specifically Nonexistent: For add-ons that collect or transmit user data, the user must be informed and provided with a clear and easy way to control this data collection. The control mechanism must be shown at first-run of the add-on. The control should contain a choice accompanied by the data collection summary. Depending on the type of data being collected, the choice to send cannot be enabled by default. If data collection starts or changes in an add-on update, or the consent and control is introduced in an update, it must be shown to all new and upgrading users. For the exact requirements, refer to https://extensionworkshop.com/documentation/publish/add-on-policies/#data-disclosure-collection-and-management . For an example of how to provide a consent and control dialog, see https://extensionworkshop.com/documentation/develop/best-practices-for-collecting-user-data-consents/ .
Also, if your add-on is listed on [addons.mozilla.org](http://addons.mozilla.org/), the listing needs to include a privacy policy, and a summary of the data collection should be mentioned in the add-on description.
Tab URL:
-&gt; src/content/index.ts line 130
-&gt; src/injected/inject.ts line 36

- Security, specifically Other security issue: Please use an approved method to sanitize remote data. You can read more about it at https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Safely_inserting_external_content_into_a_page
-&gt; src/utilities.ts line 190

Affected versions: 0.4.2

Based on that finding, those versions of your Extension have been disabled on https://addons.mozilla.org/addon/blue-blocker/ and are no longer available for download from Mozilla Add-ons, anywhere in the world. Users who have previously installed those versions will be able to continue using them.

You may upload a new version which addresses the policy violation(s).

More information about Mozilla's add-on policies can be found at https://extensionworkshop.com/documentation/publish/add-on-policies/.

Thank you for your attention.

[ref:7a7c9425-9794-495a-93bc-f03a41a48f5f]
--
Mozilla Add-ons Team
https://addons.mozilla.org/

**Comment by kheina** [comment]
I don't know what half of this means?? I could copy/paste the privacy policy from chrome into firefox's description, but idk what they mean by sanitize remote data? it doesn't inject any remote data?? not only that but the url they gave for it IS THE HTML ESCAPE FUNC???

as for these
>  -> src/content/index.ts line 130
> -> src/injected/inject.ts line 36

I assume this is just referring to the privacy policy stuff?

**Reply by kheina** [reply]
WAIT NO THE PRIVACY POLICY IS ALREADY ON FIREFOX WHAT THE HELL
![image](https://github.com/kheina-com/Blue-Blocker/assets/29378233/2fef3927-7393-4161-a730-12b5bde1ead8)

**Comment by kheina** [comment]
I left a bit of a snarky reply:

Hello, my extension already has a privacy policy (uploaded to firefox addons) and it is available here: https://addons.mozilla.org/en-US/firefox/addon/blue-blocker/privacy/

it clearly states that the addon collects cookies and header data and uses them exclusively for the main function of the addon, blocking/muting users on twitter (now x.com, hence the update)

Regarding the security issue, "sanitize remote data" my addon does not use any remote code or data except that collected from twitter and does not insert any of it into the page, again, except that collected from the same site. furthermore the code reference you sent is actually the html sanitizer used for placing text into the page.

usage example:
```
console.log(new Option("<script>dangerous code</script>").innerHTML)
// output: &lt;script&gt;dangerous code&lt;/script&gt;
```

I would appreciate it if you could re-review this version of the addon as previous versions no longer work due to twitter's url change and this update is critical to maintaining functionality.

Thanks for your time, dani

**Reply by kheina** [reply]
![image](https://github.com/kheina-com/Blue-Blocker/assets/29378233/771080cd-5ea3-4b41-a7cf-fb05d7f24874)

**Comment by kheina** [comment]
idk what to do, idk if replying like this actually triggers another manual review or not???

**Reply by crescentrose** [reply]
[The policy implies](https://extensionworkshop.com/documentation/publish/what-does-review-rejection-mean-to-users/#blocklisting) that the reviewer will receive your response and determine what to do.

**Comment by Baramanga** [comment]
Damn for w browser corpo that likes to pat itself on the back so much for being "free" and "open source" It sure has being doing a fuckton of shit to hinder users and developers in the recent years. What a fuckin shit show. I wouldn't even be surprised if the reviewers are twitter Blue users as well.

I wonder if there's a way to make this extension work through Grease Monkey 😔

**Reply by rougetimelord** [reply]
Should be possible? You would need to inject the XHR overwrite, then just parse instructions as they come in and send out block API requests. Not my cup of tea, but almost definitely feasible.

**Reply by crescentrose** [reply]
Having some sort of human oversight over code that allows you to alter almost anything in the most used app anyone uses is good actually (insert norman_rockwell_free_speech.jpg)

**Reply by PhantomGamers** [reply]
> Having some sort of human oversight over code that allows you to alter almost anything in the most used app anyone uses is good actually (insert norman_rockwell_free_speech.jpg)

The problem here is that there clearly wasn't any human oversight

**Reply by rougetimelord** [reply]
> Having some sort of human oversight over code that allows you to alter almost anything in the most used app anyone uses is good actually (insert norman_rockwell_free_speech.jpg)

it's definitely good, I just wish they would be more clear with their feedback...

**Reply by Baramanga** [reply]
> Having some sort of human oversight over code that allows you to alter almost anything in the most used app anyone uses is good actually (insert norman_rockwell_free_speech.jpg)

With the way that "feedback" was written? It's looks insanely like "we have internal reasons we won't divulge/too lazy to care so here's a standardly issued feedback". Happens all the time w Youtube unverified DMCA take down system too.

**Comment by rougetimelord** [comment]
What da hail??? 

I guess we should add ", including the URLs of pages you visit," after "API responses" in the Privacy Policy?

**Reply by cooljeanius** [reply]
...where is the Privacy Policy stored in the repo, anyways? I just looked and I don't seem to see it...

**Reply by kheina** [reply]
it's not on GitHub, it's just something I wrote when I uploaded it to Firefox initially. if you guys have any suggestions I'm happy to update

**Reply by cooljeanius** [reply]
If you could store it in the GitHub repo, that'd probably be helpful.

**Reply by kheina** [reply]
yeah, I just don't think I can get it to auto update the policy on Firefox unfortunately

**Reply by rougetimelord** [reply]
Looking at some of the bigger (corporate extensions) they have their privacy policy on AMO and a link at the top to the latest version, we could probably do that

**Reply by kheina** [reply]
they specifically say not to link to an external privacy policy. besides, I find just not collecting data the best privacy policy, so I just wrote something that made that clear.

**Reply by cooljeanius** [reply]
> yeah, I just don't think I can get it to auto update the policy on Firefox unfortunately

There's no way to get the release action to copy it over?

**Reply by kheina** [reply]
> There's no way to get the release action to copy it over?

I don't see anything in their docs about updating/uploading a privacy policy, which sucks

https://mozilla.github.io/addons-server/topics/api/addons.html#edit

you can fetch a privacy policy, but not edit one

**Reply by kheina** [reply]
there may be a hidden put/patch methods on the get endpoint for eula/privacy endpoint, but I'd need to sign a jwt to try it and I'm not feeling it at the end of the day
https://addons.mozilla.org/api/v5/addons/addon/%7B119be3f3-597c-4f6a-9caf-627ee431d374%7D/eula_policy/

**Comment by rougetimelord** [comment]
My reading was that they don't want your privacy policy on the AMO to only
be a link to an external policy, but that there isn't a flat out ban on
linking to an external policy. Sample size of one, but looking at Honey's
privacy policy they have a link at the top and then an old version of
policy.

**Comment by kheina** [comment]
(yes I'm a moron, yes a revoked those secrets immediately)
```bash
web-ext sign --api-key "user:xxxxxxxx:zz" --api-secret "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" --channel unlisted
Building web extension from /Users/dani/codebase/blue-blocker/build
Waiting for validation...
Waiting for approval...
^C
```

they require approval for only signed extensions too

**Reply by Baramanga** [reply]
So uh, did you figure it out?

**Reply by xxSeys1** [reply]
Apparently it's still in review (see last sentence of [this](https://github.com/kheina-com/Blue-Blocker/discussions/302#discussioncomment-9715639) comment in another discussion thread).

**Reply by gbootsma** [reply]
I keep wondering if Mozilla will ever approve this update. Any news?

**Reply by kheina** [reply]
likely not, @rougetimelord is working on a pr to make mozilla happy: https://github.com/kheina-com/Blue-Blocker/pull/295

**Comment by cabatchi** [comment]
it's weird because I submitted it for private distribution (sorry if this isn't allowed!) because I was tired of having to load it every time I restarted the browser, and mozilla actually approved it. version 4.2.0 btw. I don't know how.

**Reply by xxSeys1** [reply]
Wait does that mean anyone can install it as long as they are doing it via install addon from file? I just tried and I got the same error message saying that the addon isn't signed my mozilla as before

**Reply by cabatchi** [reply]
everyone has been able to install it via file for a while, unsure as to what you mean.

**Reply by cooljeanius** [reply]
> everyone has been able to install it via file for a while, unsure as to what you mean.

not everyone; I remember seeing someone who said they couldn't

**Reply by xxSeys1** [reply]
When I try to install the addon in Firefox 127.0.2 via the [Zip file for Firefox in the 4.02 release](https://github.com/kheina-com/Blue-Blocker/releases/tag/v0.4.2) a popup pops up saying that the addon is not verified by mozilla :/

**Reply by Baramanga** [reply]
> it's weird because I submitted it for private distribution (sorry if this isn't allowed!) because I was tired of having to load it every time I restarted the browser, and mozilla actually approved it. version 4.2.0 btw. I don't know how.

Can you download the addon again from Mozilla with the signature and share it?

**Reply by Staudey** [reply]
@xxSeys1 Are you using the method from the README? (https://github.com/kheina-com/Blue-Blocker?tab=readme-ov-file#firefox)

**Reply by xxSeys1** [reply]
@Staudey Nope, just downloaded the zip file, went to my addons and installed it. Is the correct way to install an addon for private distribution via installing a temporary addon? Cause I always thought those would reset as soon as I restart the browser.
Or is building the addon myself the important step?

**Comment by cooljeanius** [comment]
Just a heads up that 0.4.3 has now been released, so hopefully Firefox will accept this one.
Edit: Looks like they didn't...
